.github/workflows/rust.yml

```
@@ -10,6 +10,30 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  build-sp1:
+    runs-on: ubuntu-latest
+    needs: [cargo-fmt, cargo-clippy]
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+
+    - name: Install Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y curl git
+
+    - name: Install toolchain
+      run: |
+        curl -L https://sp1up.succinct.xyz | bash
+        source ~/.bashrc 
+        export PATH=$PATH:~/.sp1/bin
+        sp1up
```

```suggestion
        /home/runner/.sp1/bin/sp1up
```

You can simplify this like you did for risc0